mteb/models/cache_wrapper.py

```diff
@@ -0,0 +1,330 @@
+                    logger.warning(
+                        "Vector dimension not set. Unable to load vectors file."
+                    )
+
+                logger.info(
+                    f"Loaded TextVectorMap ({name_details}) from {self.directory}"
+                )
+            else:
+                logger.warning(
+                    f"No existing files found. Initialized empty TextVectorMap ({name_details})."
+                )
+        except Exception as e:
+            logger.error(f"Error loading TextVectorMap ({name_details}): {str(e)}")
+            raise
+
+    def get_vector(self, text: str) -> Optional[np.ndarray]:
+        try:
+            text_hash = self._hash_text(text)
+            if text_hash not in self.hash_to_index:
+                logger.debug(f"Text hash not found in index: {text_hash}")
+                return None
+            index = self.hash_to_index[text_hash]
+            return self.vectors[index]
+        except Exception as e:
+            logger.error(f"Error retrieving vector for text: {str(e)}")
+            raise
+
+    def __contains__(self, text: str) -> bool:
+        return self._hash_text(text) in self.hash_to_index
+
+    def __del__(self):
+        self.close()
+
+    def close(self):
+        if hasattr(self, "vectors") and self.vectors is not None:
+            self.vectors.flush()
+            del self.vectors
+            self.vectors = None
+        logger.info(f"Closed TextVectorMap in directory: {self.directory}")
+
+
+class CachedEmbeddingWrapper:
+    def __init__(self, model: Any, cache_path: Union[str, Path]):
+        self._model = model
+        self.cache_path = Path(cache_path)
+        self.cache_path.mkdir(parents=True, exist_ok=True)
+
+        if hasattr(model, "encode_queries") and hasattr(model, "encode_corpus"):
+            self.encode_method = "split"
+            self.query_cache = TextVectorMap(self.cache_path / "query_cache")
+            self.corpus_cache = TextVectorMap(self.cache_path / "corpus_cache")
+            self.query_cache.load(name="query_cache")
+            self.corpus_cache.load(name="corpus_cache")
+            self._wrap_split_encode_methods()
+        elif hasattr(model, "encode"):
+            self.encode_method = "single"
+            self.cache = TextVectorMap(self.cache_path / "cache")
+            self.cache.load(name="cache")
+            self._wrap_single_encode_method()
+        else:
+            logger.error(
+                "Model must have either 'encode_queries' and 'encode_corpus' methods, or a single 'encode' method."
+            )
+            raise ValueError("Invalid model encoding method")
+
+        logger.info(
+            f"Initialized CachedEmbeddingWrapper with {self.encode_method} encoding method"
+        )
+
+    def _wrap_split_encode_methods(self):
+        original_encode_queries = self._model.encode_queries
+        original_encode_corpus = self._model.encode_corpus
+
+        def wrapped_encode_queries(
```

Comment:
I believe we should remove these

Reply:
Sure - but does this not cause issues with custom models?

Reply:
I believe most custom models (if not all) have been refactored.

Reply:
If you about models inside models dir, I changed them all